src/Core/Managed/Shared/Extensibility/Metrics/DependencyMetricExtractor.cs

```diff
@@ -0,0 +1,169 @@
+                            new Dictionary<string, string>() {
+                                [MetricTerms.Autocollection.DependencyCall.PropertyName.Success] = Boolean.FalseString,
+                                [MetricTerms.Autocollection.DependencyCall.PropertyName.TypeName] = depTypes[i],
+                            });
+                }
+
+                this.successDurationMetricsByType = successDurationsByType;
+                this.failureDurationMetricsByType = failureDurationsByType;
+
+                this.successDurationMetricDefault = MetricManager.CreateMetric(
+                        MetricTerms.Autocollection.MetricNames.DependencyCall.Duration,
+                        new Dictionary<string, string>() {
+                            [MetricTerms.Autocollection.DependencyCall.PropertyName.Success] = Boolean.TrueString,
+                            [MetricTerms.Autocollection.DependencyCall.PropertyName.TypeName] = MetricTerms.Autocollection.DependencyCall.TypeName.Other,
+                        });
+
+                this.failureDurationMetricDefault = MetricManager.CreateMetric(
+                        MetricTerms.Autocollection.MetricNames.DependencyCall.Duration,
+                        new Dictionary<string, string>() {
+                            [MetricTerms.Autocollection.DependencyCall.PropertyName.Success] = Boolean.FalseString,
+                            [MetricTerms.Autocollection.DependencyCall.PropertyName.TypeName] = MetricTerms.Autocollection.DependencyCall.TypeName.Other,
+                        });
+            }
+        }
+
+        public override void ExtractMetrics(ITelemetry fromItem, out bool isItemProcessed)
+        {
+            DependencyTelemetry dependencyCall = fromItem as DependencyTelemetry;
+            if (dependencyCall == null)
+            {
+                isItemProcessed = false;
+                return;
+            }
+
+            bool isFailed = (dependencyCall.Success != null) && (dependencyCall.Success == false);
+            Metric metric = null;
+            string dependencyType = dependencyCall.Type;
+
+            IDictionary<string, Metric> durationMetricsByType = (isFailed)
+                                                                    ? this.failureDurationMetricsByType
+                                                                    : this.successDurationMetricsByType;
+
+            if (durationMetricsByType == null || dependencyType == null)
+            {
+                metric = (isFailed ? this.failureDurationMetricDefault : this.successDurationMetricDefault);
+            }
+            else
+            {
+                bool isKnownType = durationMetricsByType.TryGetValue(dependencyType, out metric);
+                if (! isKnownType)
+                {
+                    metric = (isFailed ? this.failureDurationMetricDefault : this.successDurationMetricDefault);
```

Comment:
this will not going to fly. Pre-configuring dependency types is not a good approach - we do not want this extra configuration problem for our customers